components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/Oauth2ScopeUtils.java

```diff
@@ -321,4 +324,44 @@ public static boolean isSystemLevelInternalSystemScopeManagementEnabled() {
         }
         return true;
     }
+
+    /**
+     * Iterate through the scopes array to filter out the internal scopes.
+     * @param scopes String array of scopes.
+     * @return String array with internal scopes. Return an empty array if there's not any internal scopes in the
+     * given scopes array.
+     */
+    public static String[] getRequestedScopes(String[] scopes) {
```

Comment:
Why do we need to have two separate methods for getRequestedScopes? can't we use the same method? We can make the the arrays as a list and send to the getRequestedScopes method right?

Reply:
Yes @nilasini. I thought it would be better if we add two different methods since converting an array to a list or list to array is a kind of time-consuming operation. If this is not a problem I'll just use one method.

Reply:
This has been fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1766/commits/ca0b0b94f26a3d0f66fc9d3e6b0ba33c7dd8075b